library.c

```diff
@@ -2126,15 +2126,20 @@ redis_pack(RedisSock *redis_sock, zval *z, char **val, strlen_t *val_len TSRMLS_
 #ifdef HAVE_REDIS_LZF
     char *data;
     uint32_t res;
+    double size;
 #endif
 
     valfree = redis_serialize(redis_sock, z, &buf, &len TSRMLS_CC);
     switch (redis_sock->compression) {
         case REDIS_COMPRESSION_LZF:
 #ifdef HAVE_REDIS_LZF
-            data = emalloc(len);
-            res = lzf_compress(buf, len, data, len - 1);
-            if (res > 0 && res < len) {
+            /**
+             * output buffer might be considerably more than in_len
+             * (but less than 104% of the original size)
+             */
+            size = ceil(len * 1.04);
+            data = emalloc(size);
+            if ((res = lzf_compress(buf, len, data, size)) > 0) {
```

Comment:
Passing a `double` into `lzf_compress` could overflow since `lzf_compress` is defined as the following:

```public int lzf_compress (byte[] in_data, int in_len,byte[] out_data, int out_len)```

Reply:
To be correct
```
unsigned int 
lzf_compress (const void *const in_data,  unsigned int in_len,
              void             *out_data, unsigned int out_len);
```

Reply:
Haha, FFS my `ctags` jumped me to the `C#` prototype (`ext/redis/liblzf/cs/CLZF.cs`) :joy:

Similar issue though, even with an unsigned int.  It's been a while since I messed with this kind of double -> int downcast, but I _think_ something like this works:

```c
// nextafter should be in limits.h
if (nextafter(size, 0) > UINT_MAX) {
    // Overflow
} else {
    // Not overflow
}
```

Now, what I for sure don't know is whether `nextafter` is called `__win64_we_are_horrible_nextfloat()` in Windows.